modern/src/RegisterDialog.js

```diff
@@ -0,0 +1,109 @@
+import t from './common/localization'
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import TextField from '@material-ui/core/TextField';
+import Alert from '@material-ui/lab/Alert';
+
+const RegisterDialog = ({ open, onResult }) => {
+  const [formFields, setFormFields] = useState({});
+  const [validationErrors, setValidationErrors] = useState({});
+  const [errorResponse, setErrorResponse] = useState(null);
+
+  const handleChange = (event) => {
+    setFormFields({ ...formFields, [event.target.name]: event.target.value });
+    setValidationErrors({ ...validationErrors, [event.target.name]: false });
+  }
+
+  const handleRegister = async (event) => {
+    event.preventDefault();
+    setErrorResponse(null);
+    let objErrors = {};
+    if (formFields.name.trim() === '') {
+      objErrors.name = true;
+    }
+    if (!
+      /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/.test(formFields.email)) {
+      objErrors.email = true;
+    }
+    if (formFields.password.trim() === '') {
+      objErrors.password = true;
+    }
+    if (Object.keys(objErrors).length !== 0) {
+      setValidationErrors(objErrors);
+      return;
+    }
+
+    const response = await fetch('/api/users', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(formFields)
+    });
+
+    if (response.ok) {
+      onResult(true)
+    } else {
+      setErrorResponse(t('errorGeneral'));
```

Comment:
I would skip this error check for now. We need more standardized error handing for all calls.

Reply:
Error checking is skipped now.